src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CreateServer.cs

```diff
@@ -254,5 +255,16 @@ public static void Windows_ServerCloneWithDifferentDirection_Throws_Unauthorized
                 Assert.Throws<UnauthorizedAccessException>(() => new NamedPipeServerStream(uniqueServerName, PipeDirection.Out));
             }
         }
+
+        [Fact]
+        public static void PipeOptions_FirstPipeInstanceWithSameNameReuse_Throws_UnauthorizedAccessException()
```

Comment:
Could you please add a test that uses `RemoteExecutor` to start a new process that tries to create a second instance of `NamedPipeServerStream` with the same name? I would like to be 100% sure that we are covering both code paths (already one instance present in the static dictionary + not present and not unlinking and failing on `Bind`).

You can use https://github.com/dotnet/runtime/blob/6f12df42691816bad57386442e405a0628392286/src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CrossProcess.cs#L48-L60 as an example and add it to `NamedPipeTest.CrossProcess.cs`.